src/notification/entities/notification.entity.ts

```diff
@@ -0,0 +1,78 @@
+import { IEntity } from 'src/shared/models/entity';
+import { Entity, Column } from 'typeorm';
+import { MailContext, NotificationType } from '../enums';
+import { NotificationSuppressedException } from '../exceptions/notification-suppressed.exception';
+
+export interface NotificationMetadata {
+  context: MailContext;
+  correlationId: string;
+}
+
+export interface NotificationOptions {
+  suppressRecurring?: boolean;
+  debounce?: number; // debounce time in milliseconds
+}
+
+@Entity()
+export class Notification extends IEntity {
+  @Column({ length: 256, nullable: false })
+  type: NotificationType;
+
+  @Column({ length: 256, nullable: false })
+  context: MailContext;
+
+  @Column({ length: 'MAX', nullable: false })
+  correlationId: string;
+
+  @Column({ type: 'datetime2', nullable: false })
+  sendDate: Date;
+
+  @Column({ nullable: false, default: false })
+  suppressRecurring: boolean;
+
+  @Column({ type: 'float', nullable: true })
+  debounce: number;
+
+  protected create(type: NotificationType, metadata?: NotificationMetadata, options?: NotificationOptions) {
+    this.sendDate = new Date();
+    this.type = type;
+
+    this.context = metadata?.context;
+    this.correlationId = metadata?.correlationId.substring(0, 255);
```

Comment:
yes, added that before changing type to MAX, thought in the end that it is not safe to rely on such thing in many places, because its easy to forget... And I was right - I forgot about this place. Will remove.